src/util/unwrapFunction.ts

```diff
@@ -0,0 +1,13 @@
+import { debugLogObject } from './debug.js';
+
+export const unwrapFunction = async (possibleFunction: unknown) => {
+  if (typeof possibleFunction === 'function') {
+    try {
+      return await possibleFunction();
+    } catch (error) {
+      debugLogObject('Error executing function:', error);
```

Comment:
Could you please rebase from the `main` branch. Just before your PR, I've added a `context` argument so it would become like so:

```suggestion
      debugLogObject('*', 'Error executing function:', error);
```

The `'*'` might look a bit odd (had to come up with something representing "global" context), but now we can "assign" debug output to workspaces.